syntax = "proto3";

package dumptool.v1;

service DumpService {
  rpc SendDump(DumpRequest) returns (DumpResponse);
}

message DumpRequest {
  string dump_path = 1;
  bytes payload = 2;
  enum DataFormat {
    JSON = 0;
    PROTOBUF = 1;
    BINARY = 2;
  }
  DataFormat format = 3;
  map<string, string> metadata = 4;
}

message DumpResponse {
  bool success = 1;
  string message = 2;
}